ml-agents/mlagents/trainers/stats.py

```diff
@@ -304,6 +343,39 @@ def __init__(self, category: str):
     def add_writer(writer: StatsWriter) -> None:
         StatsReporter.writers.append(writer)
 
+    @staticmethod
```

Comment:
StatsReporter doesn't seem like it's the right place to be saving and loading global state. Can you make a new class for this? TrainingGlobalState?

Reply:
The main reason I put it in StatsReporter was that the trainers already have one initialized with the right key (the brain_name). No problem, it can be moved